ros/src/pybullet_ros/pybullet_robot_description.py

```diff
@@ -0,0 +1,329 @@
+        # self collision is on by default
+        if rospy.get_param(namespace + "use_inertia_from_file", False):
+            urdf_flags = pb.URDF_USE_INERTIA_FROM_FILE | pb.URDF_USE_SELF_COLLISION
+        else:
+            urdf_flags = pb.URDF_USE_SELF_COLLISION
+        self._id = pb.loadURDF(urdf_path, basePosition=[pose_x, pose_y, pose_z],
+                               baseOrientation=pb.getQuaternionFromEuler([0.0, 0.0, pose_yaw]),
+                               useFixedBase=fixed_base, flags=urdf_flags, physicsClientId=self._uid)
+
+        # get joint and link info
+        self._all_joint_info = self._get_joint_info()
+        self._all_joint_names = [joint['jointName'].decode("utf-8") for joint in self._all_joint_info]
+        self._all_joint_indices = [joint['jointIndex'] for joint in self._all_joint_info]
+        self._all_joint_dict = dict(zip(self._all_joint_names, self._all_joint_indices))
+
+        self._all_link_names = [joint['linkName'].decode("utf-8") for joint in self._all_joint_info]
+        self._all_link_dict = dict(zip(self._all_link_names, self._all_joint_indices))
+
+        self._movable_joint_indices = self._find_movable_joints()
+        self._fixed_joint_indices = self._find_fixed_joints()
+
+        self._joint_lower_position_limits = [pb.getJointInfo(self._id, joint, physicsClientId=self._uid)[8] for joint in
+                                             self._movable_joint_indices]
+        self._joint_upper_position_limits = [pb.getJointInfo(self._id, joint, physicsClientId=self._uid)[9] for joint in
+                                             self._movable_joint_indices]
+        self._joint_velocity_limits = [pb.getJointInfo(self._id, joint, physicsClientId=self._uid)[11] for joint in
+                                       self._movable_joint_indices]
+        self._joint_effort_limits = [pb.getJointInfo(self._id, joint, physicsClientId=self._uid)[10] for joint in
+                                     self._movable_joint_indices]
+
+        self._joint_limits = [{'pos_lower': x[0], 'pos_upper': x[1], 'velocity': x[2], 'effort': x[3]}
+                              for x in zip(self._joint_lower_position_limits,
+                                           self._joint_upper_position_limits,
+                                           self._joint_velocity_limits, self._joint_effort_limits)]
+
+        for i, joint_id in enumerate(self._movable_joint_indices):
+            pb.resetJointState(self._id, joint_id,
+                               (self._joint_upper_position_limits[i] + self._joint_lower_position_limits[i]) / 2)
+
+        # if everything went well, the robot is ready
+        self._initialized = True
+
+    def is_initialized(self):
+        """
+        Getter of the initialized attribute.
+
+        :rtype: bool
+        """
+        return self._initialized
+
+    def get_id(self):
```

Comment:
getters like this are not very pythonic. the suggested approach for all of these simple cases (in the case that the class property should not just be public in the first place) would be:

```python
@property
def id(self):
    return self._id
```